### PR TITLE
Add clearcolor benchmark and some options for textrendering

### PR DESCRIPTION
**Pull request by necropotame**
* benchmark_clearcolor is needed to know the FPS upper bound that textrendering can reach
 * textrendring can use different set of text: ascii, monofont (the old one) and cjk

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -309,6 +309,20 @@ if(CLIENT)
     target_sources(${TARGET_CLIENT} PRIVATE "other/icons/DDNet.rc")
   endif()
   
+  # Benchmark for clear color
+  set(TARGET_BM_CLEARCOLOR benchmark_clearcolor)
```
Comment (heinrich5991) [inline CMakeLists.txt]: I didn't really know what "clear color" was when I first looked at it. Maybe call it something like "benchmark_empty"?

File: src/benchmarks/textrendering.cpp [hunk 2]
```diff
@@ -118,108 +205,69 @@ int main(int argc, const char **argv) // ignore_convention
...
+		if(str_comp(argv[i], "--help") == 0 || str_comp(argv[i], "-h") == 0)
+		{
+			dbg_msg("benchmark", "Usage: %s --test <testname>", argv[0]);
+			dbg_msg("benchmark", "Available tests: \"ascii\", \"monofont\" or \"cjk\"", argv[0]);
+			return 0;
+		}
+		else if(str_comp(argv[i], "--test") == 0 || str_comp(argv[i], "-t") == 0)
+		{
+			if(Test != TEST_DEFAULT)
+				continue;
```
Comment (heinrich5991) [inline src/benchmarks/textrendering.cpp]: Weird behavior on double-passing `--test`, but doesn't matter.

File: src/benchmarks/textrendering.cpp [hunk 3]
```diff
@@ -118,108 +205,69 @@ int main(int argc, const char **argv) // ignore_convention
...
-			else if(i == 2)
-			{
-				pSuffix = "rd";
-			}
-			str_format(aPrefix, sizeof(aPrefix), "%d%s ", i + 1, pSuffix);
-		}
-		dbg_msg("benchmark", "%slongest frame: %.2fms", aPrefix, Time);
+		case TEST_DEFAULT:
+		case TEST_MONOFONT:
+			return ExecTest(s_aaMonofont, sizeof(s_aaMonofont) / sizeof(TestString));
```
Comment (heinrich5991) [inline src/benchmarks/textrendering.cpp]: `/ sizeof(s_aaMonofont[0])` works even if the underlying type changes.

File: src/benchmarks/tools.h [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+			m_aFrameTimes.add(TimeDiff);
+			m_FrameCounter++;
+		}
+		
+		m_CurrentTime = NewTime;
+		
+		// check end of analyse
+		{
+			int64 AnalyseDuration = m_CurrentTime - m_StartTime;
+			if(AnalyseDuration > time_freq() * m_Duration)
```
Comment (heinrich5991) [inline src/benchmarks/tools.h]: Comment and the variable name: It's "analysis". English is a weird language...

File: src/benchmarks/tools.h [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+				return false;
+		}
+		
+		return true;
+	}
+	
+	void PrintReport()
+	{
+		double Time = (m_CurrentTime - m_StartTime) / (double)time_freq();
+		double Fps = (double)(m_FrameCounter - 1) / Time;
```
Comment (heinrich5991) [inline src/benchmarks/tools.h]: I think the `- 1` is only sensible if you don't count the time of the first frame.

**Comment by necropotame** [comment]
Thanks, I will fix all of that :) No problem for the time, we are all volunteers ;)

**Pull request closed without merging**